k4Interface/include/k4Interface/ITowerToolThetaModule.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2014-2023 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef RECINTERFACE_ITOWERTOOLTHETAMODULE_H
+#define RECINTERFACE_ITOWERTOOLTHETAMODULE_H
+
+// Gaudi
+#include "GaudiKernel/IAlgTool.h"
+
+// datamodel
+#include "edm4hep/CalorimeterHitCollection.h"
+#include "edm4hep/Cluster.h"
+
+/** @class ITowerToolThetaModule RecInterface/RecInterface/ITowerToolThetaModule.h ITowerToolThetaModule.h
+ *
+ *  Abstract interface to tower building tool.
+ *
+ *  @author Anna Zaborowska
+ *  @Modified by Tong Li (t.li at cern.ch), for sliding-window clustering at FCCee where the CaloTower is created from Theta-Phi
+ */
+
+class ITowerToolThetaModule : virtual public IAlgTool {
+public:
+  DeclareInterfaceID(ITowerToolThetaModule, 1, 0);
+
+  /**  Find number of calorimeter towers.
+   *   @param[out] nTheta number of towers in theta.
+   *   @param[out] nPhi number of towers in phi.
+   */
+  virtual void towersNumber(int& nTheta, int& nPhi) = 0;
```

Comment:
I assume in this case you want to modify `nTheta` and `nPhi` and that's why you are passing by reference. Related to this consider adding some `const` to some other objects that are passed by reference in case you don't want to modify them

Reply:
This is suggested by @kjvbrt according to this PR: https://github.com/key4hep/k4FWCore/pull/175
```nTheta``` and ```nPhi``` should be fixed in the clustering algorithm. Yes I will add ```const``` to this.

Reply:
Other objects passed by reference need to be modified by the tool as well.